### PR TITLE
modulesync: enable future parser tests

### PR DESCRIPTION
**Pull request by domcleal**

### CODE REVIEW

**Comment by mmoll** [comment]
rebased.

**Comment by mmoll** [comment]
Switched the discovery tests to rspec-puppet-facts. Not sure about the array vs. string thing,  but I think thats more correct like that. No idea about the IPA fails.

**Comment by ekohl** [comment]
@mmoll I think the ```$::default_ipa_server``` and ```$::default_ipa_realm``` values are not ```''```, but most likely undef.

**Comment by mmoll** [comment]
Alright, let's see what Travis is saying, but that should turn :green_heart:

**Comment by mmoll** [comment]
One 2h timeout, the rest green... as with puppet-puppet, I expect still problem with future_parser as not everything is covered by the tests, but we could just go with it for now and see what's bubbling up out there ;)

File: manifests/config.pp [hunk 1]
```diff
@@ -49,7 +49,7 @@
     class { '::foreman::config::passenger': } -> anchor { 'foreman::config_end': }
 
     if $foreman::ipa_authentication {
-      if $::default_ipa_server == '' or $::default_ipa_realm == '' {
+      if $::default_ipa_server == undef or $::default_ipa_server == '' or $::default_ipa_realm == undef or $::default_ipa_realm == '' {
```
Comment (domcleal) [inline manifests/config.pp]: feels clunky, but OK.  My only idea is: `if !$::default_ipa_server or empty($::default_ipa_server) ...`

**Comment by domcleal** [comment]
:+1:

File: manifests/config.pp (hunk 1 above)
Comment (ekohl) [inline manifests/config.pp]: Your idea is easier to read.

**Comment by ekohl** [comment]
:+1:

File: manifests/config.pp (hunk 1 above)
Comment (domcleal) [inline manifests/config.pp]: Amended.

**Comment by domcleal** [comment]
Thanks all!

**Pull request merged**